marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx

```diff
@@ -57,16 +67,31 @@ const SaveQueryModal: React.FC<Props> = (props) => {
                 props.toggleApply(false);
         }
         if(queryName.length > 0 && queryName.trim().length !== 0){
-            props.saveNewQuery(queryName, queryDescription, facets);
-            props.setSaveNewIconVisibility(false);
-            props.setSaveModalVisibility();
-
+            try {
+                await props.saveNewQuery(queryName, queryDescription, facets);
+                props.setSaveNewIconVisibility(false);
+                props.setSaveModalVisibility();
+                applySaveQuery(searchOptions.query, searchOptions.entityTypeIds, facets, queryName);
+                props.setCurrentQueryName(queryName);
+                props.setCurrentQueryDescription(queryDescription);
+            } catch (error) {
+                if (error.response.status === 400) {
```

Comment:
This is the 4th occurrence of very similar logic for handling a 400 - hopefully there's a way to reduce this duplication??